kubernetes/samples/scripts/domain-lifecycle/startCluster.sh

```diff
@@ -0,0 +1,100 @@
+# !/bin/sh
+# Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+
+script="${BASH_SOURCE[0]}"
+scriptDir="$( cd "$( dirname "${script}" )" && pwd )"
+source ${scriptDir}/helper.sh
+
+function usage() {
+
+  cat << EOF
+
+  This script starts a WebLogic cluster in a domain by patching it's
+  'serverStartPolicy' field to 'IF_NEEDED'. This change will cause the
+  operator to initiate startup of cluster's WebLogic server instance pods
+  if the pods are not already running.
+ 
+  Usage:
+ 
+    $(basename $0) -c mycluster [-n mynamespace] [-d mydomainuid] [-m kubecli]
+  
+    -c <cluster>        : Cluster name parameter is required.
+
+    -d <domain_uid>     : Domain unique-id. Default is 'sample-domain1'.
+
+    -n <namespace>      : Domain namespace. Default is 'sample-domain1-ns'.
+
+    -m <kubernetes_cli> : Kubernetes command line interface. Default is 'kubectl'.
+
+    -h                  : This help.
+   
+EOF
+exit $1
+}
+
+set -eu
+
+kubernetesCli=${KUBERNETES_CLI:-kubectl}
+clusterName=""
+domainUid="sample-domain1"
+domainNamespace="sample-domain1-ns"
+
+while getopts "c:n:m:d:h" opt; do
+  case $opt in
+    c) clusterName="${OPTARG}"
+    ;;
+    n) domainNamespace="${OPTARG}"
+    ;;
+    d) domainUid="${OPTARG}"
+    ;;
+    m) kubernetesCli="${OPTARG}"
+    ;;
+    h) usage 0
+    ;;
+    *) usage 1
+    ;;
+  esac
+done
+
+#
+# Function to perform validations, read files and initialize workspace
+#
+function initialize {
+
+  # Validate the required files exist
+  validateErrors=false
+
+  validateKubernetesCliAvailable
+  validateJqAvailable
```

Comment:
Are clearly understandable errors are generated when referencing an invalid domain-uid, a domain that does not have a corresponding introspector cm (e.g. hasn't been deployed succesfully), invalid cluster name, or invalid managed server name?   If not, then I think additional validation would be helpful.

Reply:
Ditto for the other scripts.

Reply:
$ ./startCluster.sh -d xyz -n ns-fjui -c cluster-1
Error from server (NotFound): domains.weblogic.oracle "xyz" not found
$ ./startCluster.sh -d domain1 -n xyz -c cluster-1
Error from server (NotFound): namespaces "xyz" not found

Reply:
I have added validations for cluster-name.

Reply:
LGTM